### DISCUSSION TITLE
Struggling with fundMeTest

### DISCUSSION THREAD

**Discussion by Lukelr12** [Cyfrin/foundry-full-course-cu#2619] [Q&A] [answered]
I am getting an unexpected argument error in my code. 
kinda racking my brain on going through the AI's and repositories.  
![Screenshot 2024-10-04 202648](https://github.com/user-attachments/assets/6c87bb91-12f1-4e13-aa7a-a3ab8989c732)
![Screenshot 2024-10-04 202629](https://github.com/user-attachments/assets/2aa08887-e26a-40c2-9b8f-209c24491274)

**Comment by EngrPips** [comment]
use this `forge test --mt testPriceFeedVersionAccurate --fork-url $SEPOLIA_RPC_URL -vvvv`

**Reply by Lukelr12** [reply]
same error but got further into the test 
![Screenshot 2024-10-04 210002](https://github.com/user-attachments/assets/a3dc066b-2081-47db-bda2-0759478f1bbf)

**Reply by EngrPips** [reply]
Lol this is not the same error.

**Reply by EngrPips** [reply]
Why did you pass `msg.sender` into your `FundMe` constructor in the `setUp` function? is the `msg.sender` an address of a priceFeed contract?

**Reply by Lukelr12** [reply]
because it wouldnt work without it, when i dont have msg.sender the error is "Wrong argument count for function call: 0 arguments given but expected 1"

**Reply by EngrPips** [reply]
What did Patrick pass in for his own test setup?

**Reply by Lukelr12** [reply]
this is the orignial test set up at least where i am at in the course 

'''contract FundMeTest is Test{
    FundMe fundMe;

    function setUp() external {
    fundMe = new FundMe();
    
}


    function testMinimumDollarIsFive() public view {
        assertEq(fundMe.MINIMUM_USD(), 5e18);
    }

   function testOwnerIsMsgSender() public view {
   assertEq(fundMe.i_owner(), address(this)); 
}

function testPriceFeedVersionAccurate() public view {
    uint256 priceFeed = fundMe.getVersion();
    assertEq(priceFeed, 4);
}

}
'''

**Reply by Lukelr12** [reply]
yes msg.sender is part of the constructor 

'''constructor(address priceFeed) {
        s_priceFeed = AggregatorV3Interface(priceFeed);
        i_owner = msg.sender;'''

**Reply by EngrPips** [reply, marked as answer]
Follow the 

> this is the orignial test set up at least where i am at in the course
> 
> '''contract FundMeTest is Test{ FundMe fundMe;
> 
> ```
> function setUp() external {
> fundMe = new FundMe();
> ```
> 
> }
> 
> ```
> function testMinimumDollarIsFive() public view {
>     assertEq(fundMe.MINIMUM_USD(), 5e18);
> }
> ```
> 
> function testOwnerIsMsgSender() public view { assertEq(fundMe.i_owner(), address(this)); }
> 
> function testPriceFeedVersionAccurate() public view { uint256 priceFeed = fundMe.getVersion(); assertEq(priceFeed, 4); }
> 
> } '''

This test doesn't take a constructor argument. Why is that?

**Reply by Lukelr12** [reply]
im not sure i understand what you are asking, I thought that the setUp went to the constructor.

**Reply by Lukelr12** [reply]
got it figured out thank you

**Reply by EngrPips** [reply]
I am glad you figured it out. Keep crushing it.
